tests/conftest.py

```diff
@@ -98,125 +98,198 @@ def invalid_db(
 ) -> Iterable[AstraDBCollection]:
+    """available prepopulated to each test function."""
+    writable_v_collection.truncate()
+    writable_v_collection.insert_many(VECTOR_DOCUMENTS)
+    yield writable_v_collection
+
+
+@pytest.fixture(scope="session")
+def writable_nonv_collection(db: AstraDB) -> Iterable[AstraDBCollection]:
     """
     This is lasting for the whole test. Functions can write to it,
     no guarantee (i.e. each test should use a different ID...
     """
+    collection = db.create_collection(TEST_WRITABLE_NONVECTOR_COLLECTION)
+
+    yield collection
+
+    if int(os.getenv("TEST_SKIP_COLLECTION_DELETE", "0")) == 0:
+        db.delete_collection(TEST_WRITABLE_NONVECTOR_COLLECTION)
+
+
+@pytest.fixture(scope="function")
+def empty_nonv_collection(
+    writable_nonv_collection: AstraDBCollection,
+) -> Iterable[AstraDBCollection]:
+    """available empty to each test function."""
+    writable_nonv_collection.truncate()
+    yield writable_nonv_collection
+
+
+@pytest.fixture(scope="module")
+def invalid_writable_v_collection(
+    invalid_db: AstraDB,
+) -> Iterable[AstraDBCollection]:
     collection = invalid_db.collection(
         TEST_WRITABLE_VECTOR_COLLECTION,
     )
 
     yield collection
 
 
-@pytest.fixture(scope="module")
-def readonly_vector_collection(db: AstraDB) -> Iterable[AstraDBCollection]:
-    collection = db.create_collection(
-        TEST_READONLY_VECTOR_COLLECTION,
-        dimension=2,
-    )
+@pytest.fixture(scope="function")
+def pagination_v_collection(
+    empty_v_collection: AstraDBCollection,
+) -> Iterable[AstraDBCollection]:
+    INSERT_BATCH_SIZE = 20  # max 20, fixed by API constraints
+    N = 200  # must be EVEN
 
-    collection.insert_many(VECTOR_DOCUMENTS)
+    def _mk_vector(index: int, n_total_steps: int) -> List[float]:
+        angle = 2 * math.pi * index / n_total_steps
+        return [math.cos(angle), math.sin(angle)]
 
-    yield collection
+    inserted_ids: Set[str] = set()
+    for i_batch in _batch_iterable(range(N), INSERT_BATCH_SIZE):
+        batch_ids = empty_v_collection.insert_many(
+            documents=[{"_id": str(i), "$vector": _mk_vector(i, N)} for i in i_batch]
+        )["status"]["insertedIds"]
+        inserted_ids = inserted_ids | set(batch_ids)
+    assert inserted_ids == {str(i) for i in range(N)}
 
-    db.delete_collection(TEST_READONLY_VECTOR_COLLECTION)
+    yield empty_v_collection
 
 
-@pytest.fixture
-async def async_readonly_vector_collection(
+@pytest_asyncio.fixture(scope="function")
+async def async_readonly_v_collection(
     async_db: AsyncAstraDB,
 ) -> AsyncIterable[AsyncAstraDBCollection]:
-    collection = await async_db.create_collection(
-        TEST_READONLY_VECTOR_COLLECTION,
-        dimension=2,
-    )
-
-    await collection.insert_many(VECTOR_DOCUMENTS)
+    """
+    This fixture piggybacks on its sync counterspart:
+    it must not do anything to the collection (test functions depend on both)
```

Comment:
Maybe `async_readonly_v_collection ` could have `readonly_v_collection`injection here instead of doing it in the test functions ?